lightning/src/onion_message/functional_tests.rs

```diff
@@ -539,6 +546,29 @@ fn too_big_packet_error() {
 	assert_eq!(err, SendError::TooBigPacket);
 }
 
+#[test]
+fn blinded_path_padding() {
+	// Make sure that for a blinded path, all encrypted payloads are padded to equal lengths.
+	let nodes = create_nodes(4);
+	let test_msg = TestCustomMessage::Pong;
+
+	let secp_ctx = Secp256k1::new();
+	let intermediate_nodes = [
+		ForwardNode { node_id: nodes[1].node_id, short_channel_id: None },
+		ForwardNode { node_id: nodes[2].node_id, short_channel_id: None },
+	];
+	let context = MessageContext::Custom(Vec::new());
+	let blinded_path = BlindedPath::new_for_message(&intermediate_nodes, nodes[3].node_id, context, &*nodes[3].entropy_source, &secp_ctx).unwrap();
```

Comment:
Let's add a test for payment paths, too.

Reply:
Done!
Introduce a new test for blinded payment paths in [pr3177.06](https://github.com/shaavan/rust-lightning/commits/pr3177.06)
Thanks!